service/src/main/java/io/camunda/service/license/CamundaLicense.java

```diff
@@ -7,49 +7,76 @@
  */
 package io.camunda.service.license;
 
+import io.camunda.zeebe.util.VisibleForTesting;
 import org.camunda.bpm.licensecheck.InvalidLicenseException;
 import org.camunda.bpm.licensecheck.LicenseKey;
 import org.camunda.bpm.licensecheck.LicenseKeyImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class CamundaLicense {
 
-  public String license;
-  public boolean isValid;
+  private static final Logger LOGGER = LoggerFactory.getLogger(CamundaLicense.class);
+  private boolean isValid;
+  private boolean isSelfManaged;
   private volatile boolean isInitialized;
 
-  public CamundaLicense(final String license) {
-    this.license = license;
-  }
-
-  // for testing
   public CamundaLicense() {}
 
   public boolean isValid() {
-    initializeLicenseKey();
     return isValid;
   }
 
-  protected void initializeLicenseKey() {
+  public boolean isSelfManaged() {
+    return isSelfManaged;
+  }
+
+  public void initializeWithLicense(final String license) {
     if (!isInitialized) {
       synchronized (this) {
         if (!isInitialized) {
```

Comment:
🔧 I think we can also simply synchronize the method, I doubt we have here a performance bottleneck (we of course have no data yet to see it). Thus avoids the boilerplate code and potential issues of double-checked locking see https://www.cs.umd.edu/~pugh/java/memoryModel/DoubleCheckedLocking.html